bin/builder.rs

```diff
@@ -26,7 +26,8 @@ async fn main() -> eyre::Result<()> {
     let (block_env, env_jh) = env_task.spawn();
 
     // Spawn the cache system
-    let cache_system = config.spawn_cache_system(block_env.clone());
+    let cache_tasks = CacheTasks::new(config.clone());
+    let cache_system = cache_tasks.spawn(block_env.clone());
```

Comment:
nit: this is very slightly inconsistent with other spawn functions, which take no arguments